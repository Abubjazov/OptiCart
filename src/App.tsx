import './App.css'

export const App = () => <div className='App'>Shopping cart</div>
