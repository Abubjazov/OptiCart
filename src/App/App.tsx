import { BrowserRouter, Route, Routes } from 'react-router-dom'

import { Header } from '../components'
import { CartPage, MainPage, Page404 } from '../pages'

export const App = (): JSX.Element => {
	return (
		<BrowserRouter>
			<Header />
			<Routes>
				<Route path='/optiCart/' element={<MainPage />} />
				<Route path='/optiCart/cart' element={<CartPage />} />
				<Route path='*' element={<Page404 />} />
			</Routes>
		</BrowserRouter>
	)
}
