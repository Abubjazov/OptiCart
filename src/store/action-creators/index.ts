/* eslint-disable import/no-anonymous-default-export */
import * as CartActionCreators from './cart.actionCreators'
import * as ProductActionCreators from './product.actionCreators'

export default {
	...CartActionCreators,
	...ProductActionCreators,
}
