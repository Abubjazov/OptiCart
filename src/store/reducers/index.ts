import { combineReducers } from 'redux'
import { cartReducer } from './cartReducer'
import { productReducer } from './productReducer'

export const rootReducer = combineReducers({
	product: productReducer,
	cart: cartReducer,
})

export type RootState = ReturnType<typeof rootReducer>
