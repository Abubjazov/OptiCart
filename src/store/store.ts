import { createStore } from 'redux'
import { rootReducer } from './reducers'

import { composeWithDevTools } from 'redux-devtools-extension'

const preloadedState = {}

export const store = createStore(
	rootReducer,
	preloadedState,
	composeWithDevTools()
)
