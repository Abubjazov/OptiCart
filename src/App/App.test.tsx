import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { App } from './App'
import { store } from '../store'

describe('Component: App', () => {
	test('should render App', () => {
		const { asFragment } = render(
			<Provider store={store}>
				<App />
			</Provider>
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
