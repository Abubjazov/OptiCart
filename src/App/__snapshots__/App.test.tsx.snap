// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should render App 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h1>
      sneakers. 
      <span>
        [
      </span>
       
      <a
        href="https://optimax.dev"
        rel="noreferrer"
        target="_blank"
      >
        https://optimax.dev
      </a>
       
      <span
        class="last-span"
      >
        ]
      </span>
    </h1>
    <nav
      class="header-menu"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="color: rgb(83, 181, 170);"
          >
            Main
          </a>
        </li>
        <li>
          <a
            class=""
            href="/cart"
          >
            Cart
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="main-page"
  >
    <div
      class="container"
    >
      <svg
        class="spinner"
        height="200"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width="200"
      >
        <path
          d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
          fill="none"
          stroke="#53b5aa"
          stroke-dasharray="42.76482137044271 42.76482137044271"
          stroke-linecap="round"
          stroke-width="8"
        >
          <animate
            attributeName="stroke-dashoffset"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="0;256.58892822265625"
          />
        </path>
      </svg>
    </div>
  </main>
</DocumentFragment>
`;
